ndb/src/google/cloud/ndb/model.py

```diff
@@ -2241,9 +2252,240 @@ def __init__(self, *args, **kwargs):
 
 
 class KeyProperty(Property):
-    __slots__ = ()
+    """A property that contains :class:`.Key` values.
 
-    def __init__(self, *args, **kwargs):
+    The constructor for :class:`KeyProperty` allows at most two positional
+    arguments. Any usage of :data:`None` as a positional argument will
+    be ignored. Any of the following signatures are allowed:
+
+    .. testsetup:: key-property-constructor
+
+        from google.cloud import ndb
+
+
+        class SimpleModel(ndb.Model):
+            pass
+
+    .. doctest:: key-property-constructor
+
+        >>> name = "my_value"
+        >>> ndb.KeyProperty(name)
+        KeyProperty(b'my_value')
+        >>> ndb.KeyProperty(SimpleModel)
+        KeyProperty(kind=b'SimpleModel')
+        >>> ndb.KeyProperty(name, SimpleModel)
+        KeyProperty(b'my_value', kind=b'SimpleModel')
+        >>> ndb.KeyProperty(SimpleModel, name)
+        KeyProperty(b'my_value', kind=b'SimpleModel')
```

Comment:
This `b'my_value'` / `b'SimpleModel'` is showing the same issue we've run into in the unit tests